updatecli/updatecli.d/jdk17.yaml

```
@@ -1,122 +1,132 @@
 ---
+# This YAML configuration file is used to bump the JDK17 version in various files and create a pull request with the changes.
+
 name: Bump JDK17 version
 
 scms:
   default:
     kind: github
     spec:
-      user: "{{ .github.user }}"
-      email: "{{ .github.email }}"
-      owner: "{{ .github.owner }}"
-      repository: "{{ .github.repository }}"
-      token: "{{ requiredEnv .github.token }}"
-      username: "{{ .github.username }}"
-      branch: "{{ .github.branch }}"
+      user: "{{ .github.user }}"  # GitHub user for authentication
+      email: "{{ .github.email }}"  # Email associated with the GitHub user
+      owner: "{{ .github.owner }}"  # Owner of the GitHub repository
+      repository: "{{ .github.repository }}"  # Name of the GitHub repository
+      token: "{{ requiredEnv .github.token }}"  # GitHub token for authentication
+      username: "{{ .github.username }}"  # GitHub username
+      branch: "{{ .github.branch }}"  # Branch to create the pull request on
 
 sources:
   lastVersion:
-    kind: githubrelease
-    name: Get the latest Adoptium JDK17 version
+    kind: temurin
+    name: Get the latest Adoptium JDK17 version via the API
     spec:
-      owner: "adoptium"
-      repository: "temurin17-binaries"
-      token: "{{ requiredEnv .github.token }}"
-      username: "{{ .github.username }}"
-      versionfilter:
-        kind: regex
-        # jdk-17.0.12+7(https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.12%2B7) is OK
-        # jdk-17.0.16.1+1 (https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.16.1%2B1) is OK
-        pattern: "^jdk-17.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
+      featureversion: 17  # Feature version to fetch
     transformers:
-      - trimprefix: "jdk-"
-      - replacer:
-          from: +
-          to: _
+      - trimprefix: "jdk-"  # Remove the "jdk-" prefix from the version string
 
 conditions:
-  checkTemurinAlpineRelease:
-    name: Check if the "<lastVersion>" is available for alpine
-    kind: shell
-    spec:
-      command: sh ./jdk-download-url.sh '{{source "lastVersion" }}' alpine "x64"
-    disablesourceinput: true
-  checkTemurinLinuxRelease:
-    name: Check if the "<lastVersion>" is available for Linux
-    kind: shell
-    spec:
-      command: sh ./jdk-download-url.sh '{{source "lastVersion" }}' standard "x64 aarch64 s390x"
-    disablesourceinput: true
-  checkTemurinWindowsRelease:
-    name: Check if the "<lastVersion>" is available for Windows
-    kind: shell
+  checkTemurinAllReleases:
+    name: Check if the "<lastVersion>" is available for all platforms
+    kind: temurin
+    sourceid: lastVersion  # Source to check the version from
     spec:
-      command: sh ./jdk-download-url.sh '{{source "lastVersion" }}' windows "x64"
-    disablesourceinput: true
+      featureversion: 17  # Feature version to check
+      platforms:  # List of platforms to check the version for
```

```suggestion
      featureversion: 17
      platforms:
```
No value added (same for other similar obvious comments below)